core/src/jsMain/kotlin/Observers.kt

```diff
@@ -0,0 +1,108 @@
+package com.juul.kable
+
+import com.juul.kable.external.BluetoothRemoteGATTCharacteristic
+import kotlinx.coroutines.await
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.flow
+import org.khronos.webgl.DataView
+import org.w3c.dom.events.Event
+
+private typealias EventListener = (Event) -> Unit
```

Comment:
Nit: I don't like this typealias, I'd rather just see `(Event) -> Unit` in the two places we use it.

Reply:
Being a `nit`, I'll probably keep it as-is. I generally go the `typealias` route when I need to make the high-order function nullable, it just feels like it reads better to me.

e.g. The following is a bit cluttered IMHO:

```
private fun Characteristic.createListener(): ((Event) -> Unit)? = { event ->
```

Reply:
Literally just an opinion, if you disagree feel free to totally ignore this and keep the typealias.